src/main/java/games/strategy/triplea/settings/SettingsWindow.java

```diff
@@ -1,195 +1,150 @@
-import javax.swing.border.BevelBorder;
-
-import games.strategy.engine.ClientContext;
-import games.strategy.triplea.settings.ai.AiTab;
-import games.strategy.triplea.settings.battle.calc.BattleCalcTab;
-import games.strategy.triplea.settings.battle.options.BattleOptionsTab;
-import games.strategy.triplea.settings.folders.FoldersTab;
-import games.strategy.triplea.settings.scrolling.ScrollSettingsTab;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
 import games.strategy.ui.SwingComponents;
+import swinglib.GridBagHelper;
+import swinglib.JButtonBuilder;
+import swinglib.JLabelBuilder;
+import swinglib.JPanelBuilder;
+import swinglib.JTextAreaBuilder;
+
 
 /**
- * Window that contains a tabbed panel with preference categories, each tab contains fields that allow users to update
- * game settings. The window handles generic logic around preferences, each tab will specify configuration values for
- * the settings.
- *
- * <p>
- * Overall layout:
- * </p>
- *
- * <ul>
- * <li>Primary element is a JTabbed pain, the contents are organized into rows, one row per option presented to the
- * user.</li>
- * </ul>
- *
- * <p>
- * Each row consists of: label, swing input, detailed description
- * </p>
- *
- * <p>
- * Then we have some buttons:
- * </p>
- *
- * <ul>
- * <li>default (revert) settings</li>
- * <li>save settings</li>
- * <li>close window</li>
- * </ul>
+ * UI window with controls to update game settings and preferences, {@see ClientSetting}.
+ * Settings are grouped by type, the window consists of a TabbedPane and in it we load
+ * one tab per non-hidden {@code SettingType}.
+ * All data needed to render the settings UI is pulled from the {@code ClientSetting} enum.
  */
-public class SettingsWindow extends SwingComponents.ModalJDialog {
-
-  private static final long serialVersionUID = 8108714206041495198L;
-  private static final int MAX_WIDTH = 1100;
-  private static final int TEXT_LABEL_WIDTH = MAX_WIDTH / 4;
-  private static final int ROW_HEIGHT = 60;
-
-  /**
-   * Shows the settings window. The window is modal (a user therefore cannot open multiple at a time)
-   */
-  public static void showWindow() {
-    SwingComponents.showWindow(new SettingsWindow(
-        new BattleOptionsTab(ClientContext.battleOptionsSettings()),
-        new BattleCalcTab(ClientContext.battleCalcSettings()),
-        new AiTab(ClientContext.aiSettings()),
-        new ScrollSettingsTab(ClientContext.scrollSettings()),
-        new FoldersTab(ClientContext.folderSettings())));
+enum SettingsWindow {
+  INSTANCE;
+
+  private JDialog dialog;
+
+  public synchronized void close() {
+    if (dialog != null) {
+      dialog.dispose();
+      dialog = null;
+      Arrays.stream(ClientSettingUiBinding.values())
+          .forEach(ClientSettingUiBinding::dispose);
+    }
   }
 
-  private SettingsWindow(final SettingsTab<?>... tabs) {
-    add(buildTabbedPane(tabs), BorderLayout.CENTER);
+  public synchronized void open() {
```

Comment:
We do have a shared state problem on the `dialog` variable. `synchronized` is a bit defensive, but namely as you suggested if always on the EDT thread, then not really a problem. Thank you for the good suggestion of a `Preconditons` check.